### PR TITLE
Configurable Rates <3

### PR DESCRIPTION
**Pull request by Sefiraat**
This PR allows server owners admins to set their own rates.

Default rates have been set as the pre-existing rates so as to not change current server's experiences

Changed a bit of inconsistent formatting also, though I dislike the formatting here as a whole, I didn't want to muddy this PR.

### CODE REVIEW

File: src/main/java/me/waleks/simplematerialgenerators/SMGItemSetup.java [hunk 1]
```diff
@@ -76,20 +75,21 @@ public static void setup(@Nonnull SimpleMaterialGenerators plugin) {
...
 
-        new MaterialGenerator(SMGItems.SMG_ITEM_CATEGORY,
+        new MaterialGenerator(
+            SMGItems.SMG_ITEM_CATEGORY,
             SMGItems.SMG_GENERATOR_SMOOTH_STONE,
             RecipeType.SMELTERY,
             new ItemStack[] {
                 new ItemStack(Material.SMOOTH_STONE), SlimefunItems.STEEL_INGOT, new ItemStack(Material.SMOOTH_STONE),
                 new ItemStack(Material.IRON_BARS), SMGItems.SMG_GENERATOR_SMOOTH_STONE_BROKEN,
 				new ItemStack(Material.IRON_BARS),
```
Comment (variananora) [inline src/main/java/me/waleks/simplematerialgenerators/SMGItemSetup.java]:
```suggestion
                new ItemStack(Material.IRON_BARS), SMGItems.SMG_GENERATOR_SMOOTH_STONE_BROKEN, new ItemStack(Material.IRON_BARS),
```
tabs

File: src/main/java/me/waleks/simplematerialgenerators/SMGItemSetup.java [hunk 2]
```diff
@@ -119,94 +120,96 @@ public static void setup(@Nonnull SimpleMaterialGenerators plugin) {
...
+            }).register(plugin);
+
+        new MaterialGenerator(
+            SMGItems.SMG_ITEM_CATEGORY,
+            SMGItems.SMG_GENERATOR_OBSIDIAN,
+            RecipeType.ENHANCED_CRAFTING_TABLE,
+            new ItemStack[] {
+                SlimefunItems.REINFORCED_ALLOY_INGOT, new ItemStack(Material.WATER_BUCKET), SlimefunItems.REINFORCED_ALLOY_INGOT,
+                SMGItems.SMG_GENERATOR_OBSIDIAN_BROKEN, SMGItems.SMG_GENERATOR_STONE, SMGItems.SMG_GENERATOR_OBSIDIAN_BROKEN,
+                SMGItems.SMG_GENERATOR_OBSIDIAN_BROKEN, new ItemStack(Material.LAVA_BUCKET),  SMGItems.SMG_GENERATOR_OBSIDIAN_BROKEN
```
Comment (variananora) [inline src/main/java/me/waleks/simplematerialgenerators/SMGItemSetup.java]:
```suggestion
                SMGItems.SMG_GENERATOR_OBSIDIAN_BROKEN, new ItemStack(Material.LAVA_BUCKET), SMGItems.SMG_GENERATOR_OBSIDIAN_BROKEN
```
double space

File: src/main/java/me/waleks/simplematerialgenerators/SMGItemSetup.java [hunk 3]
```diff
@@ -119,94 +120,96 @@ public static void setup(@Nonnull SimpleMaterialGenerators plugin) {
...
-                        SMGItems.SMG_GENERATOR_SMOOTH_STONE, null, SMGItems.SMG_GENERATOR_NETHERRACK
-                }).register(plugin);
+        new MaterialGenerator(
+            SMGItems.SMG_ITEM_CATEGORY,
+            SMGItems.SMG_GENERATOR_REDSTONE,
+            RecipeType.ENHANCED_CRAFTING_TABLE,
+            new ItemStack[] {
+                SMGItems.SMG_GENERATOR_REDSTONE_BROKEN, new ItemStack(Material.REDSTONE_ORE), SMGItems.SMG_GENERATOR_REDSTONE_BROKEN,
+                SMGItems.SMG_GENERATOR_REDSTONE_BROKEN, SMGItems.SMG_GENERATOR_GRAVEL, SMGItems.SMG_GENERATOR_REDSTONE_BROKEN,
+                SlimefunItems.DAMASCUS_STEEL_INGOT, SlimefunItems.DURALUMIN_INGOT,  SlimefunItems.DAMASCUS_STEEL_INGOT
```
Comment (variananora) [inline src/main/java/me/waleks/simplematerialgenerators/SMGItemSetup.java]:
```suggestion
                SlimefunItems.DAMASCUS_STEEL_INGOT, SlimefunItems.DURALUMIN_INGOT, SlimefunItems.DAMASCUS_STEEL_INGOT
```
double space

**Review by variananora — CHANGES_REQUESTED** [review]
Should we add the setting to the `config.yml` and also some comments on that?

or just let it generate automatically?

**Comment by Sefiraat** [comment]
> Should we add the setting to the `config.yml` and also some comments on that?
> 
> or just let it generate automatically?

Morning var :). As they are all item settings they will all auto-generate into items.yml :)

**Review by variananora — APPROVED** [review]
LGTM

**Pull request merged**